docsrc/allblocks.nim

```diff
@@ -1,40 +1,126 @@
 import nimoji
 import nimib
+import nimpy
+import std/[math, strutils, strformat]
 
 nbInit
 
+var nbToc: NbBlock
+
+template addToc =
+  newNbBlock("nbText", false, nb, nbToc, ""):
+    nbToc.output = "## Table of Contents:\n\n"
+
+template nbCodeBlock(name:string) =
+  let anchorName = name.toLower.replace(" ", "-")
+  nbText "<a name = \"" & anchorName & "\"></a>\n### " & name & "\n\n---"
+  # see below, but any number works for a numbered list
+  nbToc.output.add "1. <a href=\"#" & anchorName & "\">" & name & "</a>\n"
+
 nbText: """
 > This nimib document provides a brief description and example for 
 > all blocks in nimib.
 
 :warning: **This document is not finished as of now.**
 """.emojize
 
-nbText: """
-### nbCodeSkip
+addToc()
 
+nbCodeBlock:"nbCodeSkip"
+nbText: """
 Similar to `nbCode`, `nbCodeSkip` is a block that displays 
 highlighted code but does not compile or run it.
 
 Example(s):
 """
 
-nbCodeSkip: 
+nbCodeSkip:
```

Comment:
here and for all the blocks we should probably use `nimibCode` to show the nimib code when calling the blocks (and we should document this usage of nimibCode).

Reply:
The distinction between the code of the block and the displayed block is confusing. Plus, it makes the code appears twice with not much addition (the title of the block).
A sentence separating both looks better to me. I still used nimibCode in my latest commits (to come) just so that you can compare.